File: src/2024/历年真题/2024/02.cpp
#include <iostream>

int main()
{
    int a = 0, b = 1;
    int ans = 0;
    for (long long i = 1; i <= 2024041331404202; ++i)
    {
        a = (a + i) % 100;
        b = (b * i) % 100;
        if (a == b)
        {
            ans += 1;
        }
    }

    return 0;
}

File: lanqiao-review/src/2024/历年真题/2024-国赛全真模拟测试卷/上/02.py
# https://www.lanqiao.cn/problems/17113/learning/

from datetime import datetime, timedelta

start = datetime(2023, 1, 1)
end = datetime(2024, 1, 1)

ans = 0
while start != end:
    if '1' in str(start) or 0 == start.weekday():
        ans += 5
    else:
        ans += 1
    start += timedelta(days=1)

print(ans)
# print(1333)
